gammapy/estimators/flux_map.py

```diff
@@ -0,0 +1,399 @@
+    "e2dnde": ["e2dnde_err", "e2dnde_errp", "e2dnde_errn", "e2dnde_ul"],
+    "flux": ["flux_err", "flux_errp", "flux_errn", "flux_ul"],
+    "eflux": ["eflux_err", "eflux_errp", "eflux_errn", "eflux_ul"],
+    "likelihood": ["norm_err", "norm_errn", "norm_errp","norm_ul", "norm_scan", "stat_scan"],
+}
+
+log = logging.getLogger(__name__)
+
+class FluxMap(FluxEstimate):
+    """A flux map container.
+
+    It contains a set of `~gammapy.maps.Map` objects that store the estimated flux as a function of energy as well as
+    associated quantities (typically errors, upper limits, delta TS and possibly raw quantities such counts,
+    excesses etc). It also contains a reference model to convert the flux values in different formats. Usually, this
+    should be the model used to produce the flux map.
+
+    The associated map geometry can use a `RegionGeom` to store the equivalent of flux points, or a `WcsGeom`/`HpxGeom`
+    to store an energy dependent flux map.
+
+    The container relies internally on the 'Likelihood' SED type defined in :ref:`gadf:flux-points`
+    and offers convenience properties to convert to other flux formats, namely:
+    ``dnde``, ``flux``, ``eflux`` or ``e2dnde``. The conversion is done according to the reference model spectral shape.
+
+    Parameters
+    ----------
+    data : dict of `~gammapy.maps.Map`
+        the maps dictionary. Expected entries are the following:
+        * norm : the norm factor
+        * norm_err : optional, the error on the norm factor.
+        * norm_errn : optional, the negative error on the norm factor.
+        * norm_errp : optional, the positive error on the norm factor.
+        * norm_ul : optional, the upper limit on the norm factor.
+        * norm_scan : optional, the norm values of the test statistic scan.
+        * stat_scan : optional, the test statistic scan values.
+        * ts : optional, the delta TS associated with the flux value.
+        * counts : optional, the number counts value.
+    reference_model : `~gammapy.modeling.models.SkyModel`, optional
+        the reference model to use for conversions. Default in None.
+        If None, a model consisting of a point source with a power law spectrum of index 2 is assumed.
+    gti : `~gammapy.data.GTI`
+        the maps GTI information. Default is None.
+    """
+    def __init__(self, data, reference_model=None, gti=None):
+        self.geom = data['norm'].geom
+
+        if reference_model == None:
+            log.warning("No reference model set for FluxMap. Assuming point source with E^-2 spectrum.")
+            reference_model = self._default_model()
+
+        self.reference_model = reference_model
+
+        self.gti = gti
+
+        super().__init__(data, spectral_model=reference_model.spectral_model)
+
+    @staticmethod
+    def _default_model():
+        return SkyModel(spatial_model=PointSpatialModel(), spectral_model=PowerLawSpectralModel(index=2))
+
+    @property
+    def e_ref(self):
```

Comment:
For `FluxPoints` we have introduced the more explicit `energy_ref`...make it consistent here?

Reply:
Done